src/internal/partial_solution.rs

```diff
@@ -330,50 +322,68 @@ impl<DP: DependencyProvider> PartialSolution<DP> {
 
     #[cold]
     pub fn prioritized_packages(&self) -> impl Iterator<Item = (Id<DP::P>, &DP::VS)> {
-        let check_all = self.prioritize_decision_level
-            == self.current_decision_level.0.saturating_sub(1) as usize;
         let current_decision_level = self.current_decision_level;
         self.package_assignments
-            .get_range(self.prioritize_decision_level..)
+            .get_range(current_decision_level.0 as usize..)
```

Comment:
seems like we could directly iterate over `prioritized_potential_packages` and `outdated_priorities` instead of iterating over all of `package_assignments`. I think this method is only used by UV, so not sure what semantics you need.

Reply:
we use this to make prefetches, and i think currently in a kind of inefficient way. we need to emit a prefetch at least the first time we see a package, and at most when its range changes; the previous and this was good enough in profiling for now.